samcli/lib/telemetry/event.py

```diff
@@ -83,14 +84,19 @@ class Event:
 
     event_name: EventName
     event_value: str  # Validated by EventType.get_accepted_values to never be an arbitrary string
-    thread_id = threading.get_ident()  # The thread ID; used to group Events from the same command run
+    thread_id: Optional[UUID]  # The thread ID; used to group Events from the same command run
     time_stamp: str
     exception_name: Optional[str]
 
-    def __init__(self, event_name: str, event_value: str, exception_name: Optional[str] = None):
+    def __init__(
+        self, event_name: str, event_value: str, thread_id: Optional[UUID] = None, exception_name: Optional[str] = None
+    ):
         Event._verify_event(event_name, event_value)
         self.event_name = EventName(event_name)
         self.event_value = event_value
+        if not thread_id:
+            thread_id = uuid4()
```

Comment:
Should we still use `threading.get_ident()` for default value or other fields will same for that purpose?
@hawflau what do you think?

Reply:
I think `uuid4()` is good. From what I've seen empirically, `session_id` and `threading.get_ident()` are producing the same grouping effects.